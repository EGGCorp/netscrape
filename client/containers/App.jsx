import React, { Component } from 'react';
import SearchBar from '../components/SearchBar.jsx';
import SearchResultsBox from '../components/SearchResultsBox.jsx';
import getMovies from '../../api/movieFetch.js';
import Box from '@mui/material/Box';


class App extends Component {
  constructor() {
    super();
    this.state = {
      searchedTitle: 'string',
      searchResults: [],
    }
    
    this.onKeyUp = this.onKeyUp.bind(this)
  }
  
  
  async onKeyUp(e) {
    if (e.charCode === 13){
      console.log(e.target.value)
      const value = e.target.value;
      const movieList = await getMovies(value);
      console.log('movie list', movieList);
      this.setState({ searchedTitle: e.target.value }, () => {
        console.log('this is the setState', this.state.searchedTitle)
      
      })
      
    }
  } 
  
    listItems = this.searchResults.map((movie) => {
      <li key={movie.id}>movie.title</li>
    })
  
  
  render() {

    return (
      <div>
        <SearchBar onKeyUp={this.onKeyUp}/>
        <SearchResultsBox list={listItems}/>
      </div>
      
    )
  }
}

export default App;