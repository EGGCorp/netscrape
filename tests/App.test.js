import React from 'react';
import renderer from 'react-test-renderer';
// import {cleanup, fireEvent, render} from '@testing-library/react';
import App from '../client/containers/App.jsx';

test('is an App', () => {
  const component = renderer.create(
    <App></App>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  });
