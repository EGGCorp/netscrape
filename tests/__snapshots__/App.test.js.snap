// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is an App 1`] = `
<div
  id="app"
>
  <h1>
    Netscrape
  </h1>
  <div
    id="searchBarDiv"
  >
    <div
      id="search-bar"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        onKeyPress={[Function]}
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
        >
          Search movies and TV shows on Netflix
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-3"
            >
              <span>
                Search movies and TV shows on Netflix
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="mainContent"
    >
      <div
        className="searchResultList"
      />
    </div>
    <div
      className="rightSideBar"
    >
      <h2>
        Top Rated Movies
      </h2>
    </div>
    <div
      className="footer"
    />
  </div>
</div>
`;
